Userland/Libraries/LibWeb/CSS/CSSStyleSheet.cpp

```diff
@@ -138,17 +143,40 @@ void CSSStyleSheet::set_style_sheet_list(Badge<StyleSheetList>, StyleSheetList*
     m_style_sheet_list = list;
 }
 
-Optional<StringView> CSSStyleSheet::namespace_filter() const
+Optional<StringView> CSSStyleSheet::default_namespace() const
+{
+    if (m_default_namespace_rule)
+        return m_default_namespace_rule->namespace_uri().view();
+
+    return {};
+}
+
+void CSSStyleSheet::recalculate_namespaces()
 {
     for (JS::NonnullGCPtr<CSSRule> rule : *m_rules) {
-        if (rule->type() == CSSRule::Type::Namespace) {
-            auto& namespace_rule = verify_cast<CSSNamespaceRule>(*rule);
-            if (!namespace_rule.namespace_uri().is_empty() && namespace_rule.prefix().is_empty())
-                return namespace_rule.namespace_uri().view();
+        // "Any @namespace rules must follow all @charset and @import rules and precede all other
+        // non-ignored at-rules and style rules in a style sheet.
+        // ...
+        // A syntactically invalid @namespace rule (whether malformed or misplaced) must be ignored."
+        // https://drafts.csswg.org/css-namespaces/#syntax
+        switch (rule->type()) {
+        case CSSRule::Type::Import:
```

Comment:
Shouldnt `Charset` be added here too?

Reply:
We don't have `Charset` as a rule type, so this code should be okay for now :)

Reply:
Iirc, `@charset` doesn't exist as a DOM object. Just had a quick look at the specs and that seems to be right, though they've changed their minds a couple of times.